test_gpios.py

```diff
@@ -24,7 +25,13 @@ async def asyncSetUp(self):
             self.robot = await RobotClient.at_address(conf.address, opts)
         except ConnectionError:
             # There's some race condition in the Python SDK that causes
-            # reconnection to fail sometimes.
+            # reconnection to fail sometimes. See if we can figure out what
+            # the RDK server is doing that causes this trouble. We can get
+            # stack traces from any go process by sending a SIGQUIT (signal 3)
+            # which will then exit the process, but we don't want it to exit.
+            # Instead, we send a SIGUSR1 (signal 10), which should log stack
+            # traces from all goroutines but then *not* shut down the server.
+            subprocess.run(["pkill", "--signal", "10", "viam-server"])
```

Comment:
Just seeing all this for the first time. But from what I can tell of the test scripts, there is indeed only 1 viam-server running. And the script that executes must be root, so no permission issues.

Reply:
Ohh, that's unintentionally a very good catch! There are _two_ RDK servers running on these machines: a normal one on the default port (8080) for humans to use during the day, and an extra one running on port 9090 that's only used by the board canaries at night. This is going to get both servers to log stack traces. I should be looking for the one whose config is `/etc/viam-canary.json`.